### DISCUSSION TITLE
Overwrite the translations made manually by the official translations from Crowdin

### DISCUSSION THREAD

**Discussion by rbatllet** [PrestaShop/PrestaShop#41069] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Initially I have done translations manually through Prestashop because they were incomplete. There is currently a full version of the translation in my language. When upgrading to the full official translation, my manual translations are not overwritten.

### Describe the solution you'd like

Add a checkbox to be able to override all the manual translations.

**Comment by Hlavtox** [comment]
I think the best solution would be a button/hint under the translation field itself with a small button - "revert to original".

Or, for now, you can open your database, open ps_translations table and you can see all custom translations there, just delete the lines you don't need. :-)

**Comment by rbatllet** [comment]
For now, it is only allowed to reverse the translation for each string. I asked it could be done in a grouped way. Thanks for the temporary solution.

**Comment by Hlavtox** [comment]
Maybe we could do a simple table, showing all custom translations? Or a button called “delete all custom translations for this language”.

**Comment by rbatllet** [comment]
And also wouldn't it be better to add a checkbox in the place where the update button is to be able to choose to overwrite the existing ones or not?

**Comment by Hlavtox** [comment]
Well, it works like this:
- All your custom translations are saved into ps_translation table. With language ID, original string and your translation. This table could be totally empty if you are satisfied with the default translations.
- Then there are the XLF files with translations from Crowdin.
- When the code requests some translation, it first looks into the translation table for your custom translation, if it does not find it, it uses the crowdin translation from XLF file.

So, if you want to use crowdin translation, you just need to your custom one and clear the translation cache.

Now, do you really want to delete all custom translations for that language? Or single ones?

Try now to think about it, check the translation interface in backoffice for possible places to put these reset functions, think about other people, how could the probable use cases look like. Then we can polish the specs with @MatShir and implement it. 🙂

**Comment by rbatllet** [comment]
I think people who use PrestaShop want to have the updated and official versions of their supported languages from Crowdin that are constantly updated. The problem I've described, they can't be removed if you don't go to the database and do all of them manually and not everyone has access the database and the knowledge to do that.

The best place to place the reset function would be on "/international/translations/settings" page in "Add / update a language" section (you can choose only one language to update). That function should refresh the cache after removing the translations from the database.

**Comment by hibatallahAouadni** [comment]
Hello @rbatllet 

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you
